src/ducks/context/BanksContext.spec.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react'
+
+import JobsProvider from '../context/JobsContext'
+import BanksProvider, { BanksContext } from '../context/BanksContext'
+import { render } from '@testing-library/react'
+import CozyClient from 'cozy-client'
+import CozyRealtime from 'cozy-realtime'
+import { KONNECTOR_DOCTYPE } from '../../doctypes'
+
+jest.mock('cozy-realtime')
+
+export const createKonnectorMsg = (state, konnector, account) => ({
+  worker: 'konnector',
+  state,
+  message: {
+    konnector,
+    account
+  }
+})
+
+const RUNNING = 'running'
+const KONNECTORS = [
+  { konnector: 'caissedepargne1', account: '1234' },
+  { konnector: 'boursorama83', account: '5678' }
+]
+
+describe('Jobs Context', () => {
+  const setup = ({ withKonnectors }) => {
+    const client = new CozyClient({})
+    client.query = jest.fn().mockImplementation(options => {
+      const { doctype } = options
+      if (doctype === KONNECTOR_DOCTYPE) {
+        return {
+          data: KONNECTORS.map(k => ({
+            account: k.account,
+            slug: k.konnector
+          }))
+        }
+      }
+    })
+    CozyRealtime.mockImplementation(() => {
+      return {
+        subscribe: (eventName, doctype, handleRealtime) => {
+          // There are 3 subscribers (created, updated, deleted)
+          // To simulate handle realtime we check if there are
+          // at least the first event and we call handleRealtime callbacks
+          if (eventName === 'created') {
+            if (withKonnectors) {
+              handleRealtime(
+                createKonnectorMsg(
+                  RUNNING,
+                  KONNECTORS[0].konnector,
+                  KONNECTORS[0].account
+                )
+              )
+              handleRealtime(
+                createKonnectorMsg(
+                  RUNNING,
+                  KONNECTORS[1].konnector,
+                  KONNECTORS[1].account
+                )
+              )
```

Comment:
Nit: I think it'd be a bit clearer if konnectors could be passed to the setup method instead of konnectors.

```
// here 
          if (eventName === 'created') {
            for (let konn of konnectors) {
              handleRealtime(
                createKonnectorMsg(
                  RUNNING,
                  konn.konnector,
                  konn.account
                )
            }
              
         }

// in the tests
const { root } = setup({ konnectors: [konnectors[0], konnectors[1]] })
```